tools/mo/unit_tests/mo/front/common/partial_infer/eltwise_test.py

```diff
@@ -105,3 +109,68 @@ def test_eltwise_infer_none_min_max(self):
+
+        nodes = {
+            **shaped_parameter('undefined_shape_data', None, {'reverse_infer': Parameter.reverse_infer}),
+            **shaped_parameter('data', shape_array(defined_shape), {'reverse_infer': Parameter.reverse_infer}),
+            **regular_op_with_empty_data('elementwise', {'op': 'Add', 'type': 'Add',
+                                                         'infer': eltwise_infer,
+                                                         'reverse_infer': eltwise_reverse_infer}),
+            **result('res'),
+        }
+
+        edges = [
+            *connect('undefined_shape_data', '{}:elementwise'.format(int(not in_port_with_defined_shape))),
+            *connect('data', '{}:elementwise'.format(in_port_with_defined_shape)),
+            *connect('elementwise', 'res')
+        ]
+
+        graph = build_graph(nodes, edges)
+        graph.stage = 'middle'
+        Node(graph, 'elementwise').out_port(0).data.set_shape(shape_array(out_shape))
+        Node(graph, 'elementwise').in_port(in_port_with_defined_shape).data.set_shape(defined_shape)
+
+        partial_infer(graph)
+        actual_shape = Node(graph, 'undefined_shape_data').out_port(0).data.get_shape()
+        assert strict_compare_tensors(actual_shape, shape_array(ref_shape))
+
+
+    def test_reverse_infer_1(self):
+        self.build_and_test_reverse_inference(inp_shape_1=[dyn, dyn],
+                                              inp_shape_2=None,
+                                              out_shape=[dyn, dyn, dyn, dyn],
+                                              ref_shape=[dyn, dyn, dyn, dyn])
+
+    def test_reverse_infer_2(self):
+        self.build_and_test_reverse_inference(inp_shape_1=None,
+                                              inp_shape_2=[dyn, dyn],
+                                              out_shape=[dyn, dyn, dyn, dyn],
+                                              ref_shape=[dyn, dyn, dyn, dyn])
+
+    def test_reverse_infer_3(self):
+        self.build_and_test_reverse_inference(inp_shape_1=[1],
+                                              inp_shape_2=None,
+                                              out_shape=[dyn, 400, 400, 3],
+                                              ref_shape=[dyn, 400, 400, 3])
+
+    def test_reverse_infer_4(self):
+        self.build_and_test_reverse_inference(inp_shape_1=[4, 1],
+                                              inp_shape_2=None,
+                                              out_shape=[dyn, dyn, 4, 3],
+                                              ref_shape=[dyn, dyn, dyn, 3])
+
+    def test_reverse_infer_5(self):
+        self.build_and_test_reverse_inference(inp_shape_1=[4, 1],
+                                              inp_shape_2=None,
+                                              out_shape=[dyn, dyn, 4, 1],
+                                              ref_shape=[dyn, dyn, dyn, dyn])
```

Comment:
also please add test cases when rank/shape deduction is not possible

Reply:
Agree. Added checks if deducing is possible and corresponding unit-tests

https://github.com/pavel-esir/openvino/blob/reverse_infer_extended/tools/mo/unit_tests/mo/front/common/partial_infer/eltwise_test.py#L181-L201